Account

julien-vendran opened the pull request:
Adding the part of the management for the new account

julien-vendran committed:
Adding sign in part to front

julien-vendran committed:
Adding sign in part to front (navbar)

julien-vendran committed:
Add login verification for "Inscription" button
Some design update

julien-vendran committed:
Modification du message d'erreur pour le rendre plus compréhensible par le client.

julien-vendran committed:
Modification du message d'erreur.

julien-vendran committed:
Quelques petits changements.

julien-vendran committed:
Ajout de la page validation à la navbar
+ disparition de l'option si ce n'est pas un admin
+ Ajout de la page de gestion des utilisateurs (Accepter / refuser les nouveaux)

BrokenSwing commented on components/AskResetPassword.vue:
```diff
@@ -75,9 +75,10 @@ export default {
           .catch(() => {
             this.$buefy.toast.open({
-              message: "Impossible de se connecter au serveur ❌",
+              message: "Une erreur est survenue, votre compte n'a pas été validé ❌",
```
The catch callback is triggered only when the client can't reach the server. If a well formed email is sent to the server, the server will always respond that an email was sent. Therefore, the message "Impossible de se connecter su serveur" is correct, where the part "votre compte n'a pas été validé" isn't. The reset password functionnality shouldn't be able to be used to do email discovery

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+    <tbody>
+    <NewUser
+      v-for="(user, index) in allUsers"
+      v-bind:key="`members-${index}`"
```
Use `:key="user.id"`, it's a way more stable id. Moreover, you don't need to prepend the id with `members-` as keys are unique in the same parent DOM element.

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+  apollo: {
+    allUsers: gql`{
+      allUsers {
+        firstname,lastname,roles,email
```
Why do you need to retrieve user roles ?

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+  methods: {
+    async update() {
+      console.log("update");
+      console.log(this.allUsers);
```
Remove

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+
+  methods: {
+    async update() {
+      console.log("update");
```
Remove

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+          }
+        }`
+      ).then(r => this.allUsers = r);
+      console.log(this.allUsers);
```
Remove

BrokenSwing commented on components/BuilderNewUser.vue:
```diff
@@ -0,0 +1,72 @@
+  },
+
+  methods: {
+    async update() {
```
Where is this method used ?

BrokenSwing commented on components/NewUser.vue:
```diff
@@ -0,0 +1,76 @@
+  },
+  methods: {
+    accept () {
+      console.log(this.mail);
```
Remove

BrokenSwing commented on components/SignIn.vue:
```diff
@@ -0,0 +1,162 @@
+import InputField from '~/components/InputField';
+import gql from 'graphql-tag';
+export default {
+  name: 'SignIn',
```
Did you mean SignUp ?

BrokenSwing commented on components/Navbar.vue:
```diff
@@ -1,7 +1,10 @@
 <template>
```
I'll rework navbar later, it's a complete mess 😄

BrokenSwing commented on components/NewUser.vue:
```diff
@@ -0,0 +1,76 @@
+  methods: {
+    accept () {
+      console.log(this.mail);
+      if (confirm("Vous êtes sur le point de valider " + this.firstName + " " + this.lastName + "\n Voulez-vous confirmer ?")) {
```
`confirm` ? Really ? Actually I would suggest a checkable table https://buefy.org/documentation/table#checkable that allows to batch acceptation/removal. Take this comment as a global comment on `NewUser.vue` and `BuilderNewUser.vue`

BrokenSwing replied:
Otherwise, an other solution is to keep your first setup and use https://buefy.org/documentation/dialog/#confirm instead of native JS `confirm`

BrokenSwing commented on components/NewUser.vue:
```diff
@@ -0,0 +1,76 @@
+      }
+    },
+    decline () {
+      console.log(this.mail);
```
Remove

BrokenSwing commented on components/SignIn.vue:
```diff
@@ -0,0 +1,162 @@
+          :validator='lastNameValidator'
+          required
+        />
+
```
We also want to know it's specialty (name and year), and BDE (assume BDE is Montpellier)

BrokenSwing commented on pages/validationMembres.vue:
```diff
@@ -0,0 +1,48 @@
+    };
+  },
+  methods: {
+    checkUserRoles(usersRoles) {
```
If defined, you should use it

BrokenSwing commented on static/website.json:
```diff
@@ -497,5 +497,22 @@
       ]
     }
+  ],
+  "test" : [
```
This has nothing to do in pull request

BrokenSwing reviewed (changes requested):
Lots of things to modify / rework. Do not hesitate to use Buefy components for tables

julien-vendran committed:
First update for the pull request.

julien-vendran committed:
Update for the validationMembre page.
Admin can select multiple users to validate them.
Usage of the buefy technology

julien-vendran committed:
Update navbar + del all team page and components
